### ISSUE TITLE
[Question] Can You Make VSTS Agent A System Service On OS X?

### ISSUE THREAD

**Issue by andrewtechhelp** [Microsoft/vsts-agent#176]
Currently, when you configure a VSTS agent to run as a service on Mac OS X, it makes it a user service, meaning that the service only runs when that specific user is logged in.

Is it possible to have this configured to run as a system service, so that once the Mac is started, the VSTS agent starts up without having to log into a specific user account? (or at least add this as an option). This would make it much more appropriate for non-interactive build servers.

The current non-preview agent on Windows actually runs as a system service and is available without logging in, which is perfect for build servers.

**Comment by bryanmacfarlane** [comment]
Yes, it's very possible.  It's just that our config convenience does it as a user service.

Our previous vso-agent offered both options.  Turns out the vast majority of our customers (Xamarin iOS/xcodebuild) need UI access for UI tests.  While LaunchDaemons were the default it caused problems for most users in most scenarios.  So, our convenience does what most users do.

We have a template and it's simple to create a launchdaemon if that's what you want.  Simply create a file, set some permissions and use launchctl cmd line.

I updated the docs here:

https://github.com/Microsoft/vsts-agent/blob/master/docs/start/nixsvc.md
